docker/reference/reference.go

```diff
@@ -65,7 +66,11 @@ func ParseNamed(s string) (Named, error) {
 		return nil, err
 	}
 	if canonical, isCanonical := named.(distreference.Canonical); isCanonical {
-		return WithDigest(r, canonical.Digest())
+		r, err := distreference.WithDigest(r, digest)
```

Comment:
Sorry to be a pest, but this seems incorrect (i.e. reverting this commit would lead to breakage).

Also, _some_ record of why we are doing this, for posterity, would be nice (perhaps just a link to https://github.com/docker/distribution/issues/2130 in the commit message),